File: pythonService/config/redis_client.py
import json
import redis
import os
from dotenv import load_dotenv
import logging

logger = logging.getLogger(__name__)

load_dotenv()

REDIS_HOST = os.getenv("REDIS_HOST")
REDIS_PORT = int(os.getenv("REDIS_PORT", "6379"))
REDIS_USERNAME = os.getenv("REDIS_USERNAME")
REDIS_PASSWORD = os.getenv("REDIS_PASSWORD")


redis_client = redis.Redis(
    host=REDIS_HOST,
    port=REDIS_PORT,
    username=REDIS_USERNAME,
    password=REDIS_PASSWORD,
    decode_responses=True,
)

def check_redis_connection():
  try:
    result = redis_client.ping()
    logger.info(f"[Redis]: Connected successfully - ping={result}")
  except Exception as e:
    logger.exception(f"[Redis]: Connection failed: {e}")
    raise

File: src/config/redisClient.js
// redisClient.js
import { createClient } from "redis";
import dotenv from "dotenv";

dotenv.config();

const redisOptions = {
  username: process.env.REDIS_USERNAME,
  password: process.env.REDIS_PASSWORD,
  socket: {
    host: process.env.REDIS_HOST,
    port: Number(process.env.REDIS_PORT),
  },
};

export const redis = createClient(redisOptions);

redis.on("error", (err) => console.error("[Redis]: Redis Client Error", err));

export const connectRedis = async () => {
  if (!redis.isOpen) {
    await redis.connect();
  }

  const result = await redis.ping();
  console.log("[Redis]: Redis Connected -", result);
};

export const createRedisAdapterClients = async () => {
  const pubClient = createClient(redisOptions);
  const subClient = pubClient.duplicate();

  pubClient.on("error", (err) =>
    console.error("[Redis Adapter]: Publisher error", err),
  );
  subClient.on("error", (err) =>
    console.error("[Redis Adapter]: Subscriber error", err),
  );

  await pubClient.connect();
  await subClient.connect();

  return { pubClient, subClient };
};

export const createRedisSubscriber = async () => {
  const subscriber = createClient(redisOptions);

  subscriber.on("error", (err) => {
    console.error("[Redis Subscriber]: Error", err);
  });

  await subscriber.connect();
  return subscriber;
};
